### PR TITLE
Setup Associations

### PR DESCRIPTION
**Pull request by ankitt26**
- Create a class `Classroom` 
   - create `@label` instance variable 
   - create Setter and getter for `@label`
   - create the `has-many/belongs-to` relationship between `Classroom` and `Student`.
- Create a class `Book`
   - create `@title` and `@author` instance variables
   - create Setters and getters for `@title` and `@author`.
- Create a class `Rental` 
   - create `@date` instance variable
   - create Setter and getter for `@date `

- Create the `many-to-many `(also has-many-through) relationship between `Person` and `Book` 
    using the intermediate class `Rental`.
  - Create the `has-many `side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person)
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+require_relative 'classroom'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    @classroom.add_student(self) if classroom
   end
```
Comment (abdulhamiid) [inline student.rb]: Please add a `method` that serves as the 'belongs-to' side (a student belongs to a classroom)

As you have created the `has-many` side (a classroom has many students). 

Please create the `belongs-to` side (a student belongs to a classroom) in `student.rb`. i.e (Both ways relationships)

```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```



Link to the article with examples: [Here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (abdulhamiid) [inline person.rb:13]: - Please create the `has-many` side of `Person` and `Rental` (a person has many rentals).
- A method for adding rentals should be present 👍🤓
- According to the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md), The person itself should be sent as a parameter to create the new rental achieving the `has-many` association

```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def initialize(age, name = "Unknown", parent_permission = true)
    .
    .
    @rentals = []
  end

  def add_rental(...)
  end

  .
  .
end
```

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @ankitt26 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Run a program using the command line.
- ✔️ Set up associations between classes and object

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (ankitt26) [reply student.rb]: Thank you for bringing this to my attention. I am updating the code.

File: person.rb (hunk 2 above)
Comment (ankitt26) [reply person.rb:13]: Thank you for bringing this to my attention. I am updating the code 🙂.

**Review by sumairq — APPROVED** [review]
### Hi @ankitt26 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ankitt26** [comment]
Thanks for review😊.

**Pull request merged**